### PR TITLE
Puts a cap on the size of Water-Potassium explosions

### PR DESCRIPTION
**Pull request by Coolrune206**
## What Does This PR Do

Limits the size of water-potassium explosions to "30 TNT". Previously, it had no cap. See "Images of changes" for the difference this makes.
## Why It's Good For The Game

We have a whole host of cool chemicals to blow shit up like black powder, nitroglycerine, but none of it matters if the chemicals you can pour out of a chemical dispenser straight can still make explosions nearly as large or just as large simply by using larger beakers. Making massive explosions should take either actually using chemistry to make the cooler explosion chems or toxins to make toxin bombs (which have their own downsides). Water + Potassium grenades will still not be _terrible_ for causing significant damage, but if you want to run around with a bunch of pocket maxcaps you'll have to make the more complicated chemicals.
## Images of changes

Before: (300 water, 300 potassium, large grenade) Explosion size was (4, 10, 20, 0)
image

After: (300 water, 300 potassium, large grenade) Explosion size was (1, 4, 9, 0)
image
## Testing

See above.
## Changelog
:cl:
tweak: Added a cap on the size of Water-Potassium reactions.
/:cl:

### CODE REVIEW

**Comment by ATP-Engineer** [comment]
image

**Comment by RagingLemons1** [comment]
Insert whining here about nitro being botany locked.

**Comment by Coolrune206** [comment]
> Insert whining here about nitro being botany locked.

Good, cooperate with other departments or create a solution to make your own.

File: code/modules/reagents/chemistry/recipes/pyrotechnics.dm [hunk 1]
```diff
@@ -8,7 +8,7 @@
 
 /datum/chemical_reaction/explosion_potassium/on_reaction(datum/reagents/holder, created_volume)
 	var/datum/effect_system/reagents_explosion/e = new()
-	e.set_up(round (created_volume/10, 1), holder.my_atom, 0, 0)
+	e.set_up(min(round (created_volume/10, 1), 30), holder.my_atom, 0, 0)
```
Comment (Sirryan2002) [inline code/modules/reagents/chemistry/recipes/pyrotechnics.dm:11]:
```suggestion
	e.set_up(min(round(created_volume / 10, 1), 30), holder.my_atom, 0, 0)
```

**Comment by lewcc** [comment]
Feels like it'd make sense to change the scaling of this reaction too, instead of just putting a hard cap on it? Seems a little odd that at some point you just won't get any return for mixing more.

For the record though, full agree with this. It's insane how easy it is to make a maxcap with two round-start chems.

**Comment by tipoima** [comment]
How much water+posassium does the cap correspond to? i.e. what's the amount beyond which bomb won't get stronger?

**Review by S34NW — APPROVED** [review]

**Pull request merged**